File: app/src/main/java/com/absinthe/kage/adapter/DeviceAdapter.java
package com.absinthe.kage.adapter;

import androidx.annotation.NonNull;

import com.absinthe.kage.R;
import com.absinthe.kage.device.model.DeviceInfo;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

public class DeviceAdapter extends BaseQuickAdapter<DeviceInfo, BaseViewHolder> {

    public DeviceAdapter() {
        super(R.layout.item_device);
        addChildClickViewIds(R.id.btn_connect);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder baseViewHolder, DeviceInfo deviceInfo) {
        baseViewHolder.setText(R.id.tv_device_name, deviceInfo.getName());
        baseViewHolder.setText(R.id.tv_device_ip, deviceInfo.getIp());
        switch (deviceInfo.getState()) {
            case DeviceInfo.STATE_IDLE:
                baseViewHolder.setText(R.id.btn_connect, "Connect");
                break;
            case DeviceInfo.STATE_CONNECTING:
                baseViewHolder.setText(R.id.btn_connect, "Connecting");
                break;
            case DeviceInfo.STATE_CONNECTED:
                baseViewHolder.setText(R.id.btn_connect, "Connected");
                break;
            default:
        }
    }
}

File: app/src/main/java/com/absinthe/kage/adapter/DeviceAdapter.kt
package com.absinthe.kage.adapter

import com.absinthe.kage.R
import com.absinthe.kage.device.model.DeviceInfo
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder

open class DeviceAdapter : BaseQuickAdapter<DeviceInfo, BaseViewHolder>(R.layout.item_device) {

    override fun convert(helper: BaseViewHolder, item: DeviceInfo) {
        helper.setText(R.id.tv_device_name, item.name)
        helper.setText(R.id.tv_device_ip, item.ip)

        when (item.state) {
            DeviceInfo.STATE_IDLE -> helper.setText(R.id.btn_connect, R.string.connect_state_connect)
            DeviceInfo.STATE_CONNECTING -> helper.setText(R.id.btn_connect, R.string.connect_state_connecting)
            DeviceInfo.STATE_CONNECTED -> helper.setText(R.id.btn_connect, R.string.connect_state_connected)
        }
    }

    init {
        addChildClickViewIds(R.id.btn_connect)
    }
}